app-text/poppler/poppler-0.63.0.recipe

```diff
@@ -0,0 +1,433 @@
+ 	2017 Sebastian Rasmussen
+ 	2009 Shawn Rutledge
+ 	2009 Shen Liang
+ 	2011 Simon Kellner
+ 	2010 Srinivas Adicherla
+ 	2005-2006 Stefan Kebekus
+ 	2006 Stefan Schweizer
+ 	2009 Stefan Thomas
+ 	2011 Stephen Reichling
+ 	2008 Steven G. Johnson
+ 	2014 Steven Lee
+ 	2011 Steven Murdoch
+ 	2010, 2012-2013, 2015 Suzuki Toshiya
+ 	2015 Tamas Szekeres
+ 	2005-2006 Takashi Iwai
+ 	2016 Takahiro Hashimoto
+ 	The GLib Team
+ 	2005-2010 The Poppler Developers
+ 	2013 Thomas Fischer
+ 	2009-2017 Thomas Freitag
+ 	2011 Thomas Klausner
+ 	2006 Thorkild Stray
+ 	2009, 2014 Till Kamppeter
+ 	2008 Timothy Lee
+ 	2017 Tobias C. Berner
+ 	2005, 2012, 2015 Tobias Koening
+ 	2011 Tom Gleason
+ 	2008 Tomas Are Haavet
+ 	2010-2011 Tomas Hoger
+ 	2016 Tor Lillqvist
+ 	2008 Vasile Gaburici
+ 	2016 Vincent Le Garrec
+ 	2011 Vittal Aithal
+ 	2012 Yi Yang
+ 	2013 Yury G. Kudryashov
+ 	2008-2009 Warren Toomey
+ 	2006 Wilfried Huss
+ 	2009-2016 William Bader
+ 	2007 Will Stephenson
+ 	1994 X Consortium"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://poppler.freedesktop.org/poppler-$portVersion.tar.xz"
+CHECKSUM_SHA256="27cc8addafc791e1a26ce6acc2b490926ea73a4f89196dd8a7742cff7cf8a111"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="74.0.0"
+libCppVersion="0.4.0"
+libGlibVersion="8.9.0"
+libQt5Version="1.13.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+libCppVersionCompat="$libCppVersion compat >= ${libCppVersion%%.*}"
+libGlibVersionCompat="$libGlibVersion compat >= ${libGlibVersion%%.*}"
+libQt5VersionCompat="$libQt5Version compat >= ${libQt5Version%%.*}"
+
+PROVIDES="
+	poppler$secondaryArchSuffix = $portVersion
+	lib:libpoppler$secondaryArchSuffix = $libVersionCompat
+	lib:libpoppler_cpp$secondaryArchSuffix = $libCppVersionCompat
+	cmd:pdfdetach
+	cmd:pdfimages
+	cmd:pdfseparate
+	cmd:pdftohtml
+	cmd:pdftops
+	cmd:pdfunite
+	cmd:pdffonts
+	cmd:pdfinfo
+	cmd:pdftocairo
+	cmd:pdftoppm
+	cmd:pdftotext
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	poppler_data
+	lib:libcairo$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:liblcms2$secondaryArchSuffix
+	lib:libopenjp2$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	poppler${secondaryArchSuffix}_devel = $portVersion
+	devel:libpoppler$secondaryArchSuffix = $libVersionCompat
+	devel:libpoppler${secondaryArchSuffix}_cpp = $libCppVersionCompat
+	devel:libpoppler${secondaryArchSuffix}_glib = $libGlibVersionCompat
+	devel:libpoppler${secondaryArchSuffix}_qt5 = $libQt5VersionCompat
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
+	poppler$secondaryArchSuffix == $portVersion base
+	lib:libpoppler${secondaryArchSuffix}_glib
+	lib:libpoppler${secondaryArchSuffix}_qt5
```

Comment:
```
	lib:libpoppler${secondaryArchSuffix}_glib
	lib:libpoppler${secondaryArchSuffix}_qt5
```
I guess these should not be in `REQUIRES_devel` but in `PROVIDES`, otherwise you would not have their `devel:` counterparts in `PROVIDES_devel`.
That said, I might be missing something because I did not test the recipe.

Reply:
Actually devel:libpoppler${secondaryArchSuffix}_glib and devel:libpoppler${secondaryArchSuffix}_qt5 are usually symlinks to shared libraries, which need to be installed so that the devel symlinks can work. Hence the presence of lib:libpoppler${secondaryArchSuffix}_glib and lib:libpoppler${secondaryArchSuffix}_qt5 in REQUIRES_devel.

Reply:
I think they should be

```diff
-	lib:libpoppler${secondaryArchSuffix}_glib
-	lib:libpoppler${secondaryArchSuffix}_qt5
+	poppler_glib$secondaryArchSuffix == $portVersion
+	poppler_qt5$secondaryArchSuffix == $portVersion
```

instead.